templates/arm-aks-template.json

```diff
@@ -142,13 +142,6 @@
           {
             "name": "[variables('nodepoolName')]",
             "count": "[int(parameters('nodeMinCount'))]",
-            "maxCount": "[if(empty(parameters('nodeMaxCount')), json('null') ,int(parameters('nodeMaxCount')))]",
```

Comment:
We don't have to remove these lines at all from the ARM template. If the variables are empty strings this will automatically be null. It is already tested with both autoscaler on and off.